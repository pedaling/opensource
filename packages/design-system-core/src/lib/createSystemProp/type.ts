import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
