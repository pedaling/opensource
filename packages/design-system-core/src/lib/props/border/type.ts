import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
