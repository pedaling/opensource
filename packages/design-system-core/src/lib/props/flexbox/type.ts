import type { ResponsiveValue } from '../../types';

export type FlexboxProps = {
  flex?: ResponsiveValue<string | number>;
  flexGrow?: ResponsiveValue<string | number>;
  flexShrink?: ResponsiveValue<string | number>;
  flexBasis?: ResponsiveValue<string | number>;
  flexDirection?: ResponsiveValue<'column' | 'column-reverse' | 'row' | 'row-reverse'>;
  flexWrap?: ResponsiveValue<'nowrap' | 'wrap' | 'wrap-reverse'>;
  alignContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'baseline'
    | 'normal'
  >;
  alignItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  alignSelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  justifyContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'left'
    | 'normal'
    | 'right'
  >;
  justifyItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'left'
    | 'legacy'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  justifySelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'left'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  order?: ResponsiveValue<number>;
};
