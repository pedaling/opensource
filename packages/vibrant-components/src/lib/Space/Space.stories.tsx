import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Box } from '@vibrant-ui/core';
import { VStack } from '../VStack';
import { Space } from './Space';

export default {
  title: 'Space',
  component: Space,
  args: {
    height: 20,
  },
} as ComponentMeta<typeof Space>;

export const Basic: ComponentStory<typeof Space> = props => (
  <VStack>
    <Box width={200} height={200} backgroundColor="primary" />
    <Space {...props} />
    <Box width={200} height={200} backgroundColor="primary" />
  </VStack>
);
