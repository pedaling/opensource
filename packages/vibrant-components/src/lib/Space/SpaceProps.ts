import type { DisplayProps, FlexboxProps, SizingProps } from '@vibrant-ui/core';
import { withVariation } from '@vibrant-ui/core';

export type SpaceProps = Pick<DisplayProps, 'hidden'> &
  (
    | (Pick<FlexboxProps, 'flex'> & {
        width?: never;
        height?: never;
      })
    | (Pick<SizingProps, 'height'> & {
        flex?: never;
        width?: never;
      })
    | (Pick<SizingProps, 'width'> & {
        flex?: never;
        height?: never;
      })
  );

export const withSpaceVariation = withVariation<SpaceProps>()();
