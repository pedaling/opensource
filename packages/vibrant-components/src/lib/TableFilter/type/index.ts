export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
