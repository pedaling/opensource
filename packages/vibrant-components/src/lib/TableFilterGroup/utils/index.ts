import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
