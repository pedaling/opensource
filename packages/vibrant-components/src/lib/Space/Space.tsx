import { Box } from '@vibrant-ui/core';
import { withSpaceVariation } from './SpaceProps';

export const Space = withSpaceVariation(props => <Box display="flex" {...props} />);
