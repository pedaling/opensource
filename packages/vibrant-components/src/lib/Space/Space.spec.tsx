import type { ReactRenderer } from '@vibrant-ui/utils/testing';
import { createReactRenderer } from '@vibrant-ui/utils/testing';
import { Space } from './Space';

describe('<Space />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;
  let element: HTMLElement;

  describe('when Space rendered', () => {
    beforeEach(() => {
      renderer = render(<Space data-testid="space" width={20} />);

      element = renderer.getByTestId('space');
    });

    it('width style applied', () => {
      expect(element).toHaveStyleRule('width', '20px');
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when hidden Space rendered', () => {
    beforeEach(() => {
      renderer = render(<Space data-testid="space" visible={false} />);

      element = renderer.getByTestId('space');
    });

    it('space component is hidden', () => {
      expect(element).toHaveStyleRule('display', 'none');
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
