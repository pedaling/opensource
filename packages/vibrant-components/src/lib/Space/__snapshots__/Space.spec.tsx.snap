// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Space /> when Space rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: 20px;
}

<div>
  <div
    class="emotion-0"
    data-testid="space"
  />
</div>
`;

exports[`<Space /> when hidden Space rendered match snapshot 1`] = `
.emotion-0 {
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<div>
  <div
    class="emotion-0"
    data-testid="space"
  />
</div>
`;
